### PR TITLE
[python-updates] python3Packages.tensorflow: 2.7.1 -> 2.8.0

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change
As part of `python-updates` a `tensorflow-estimator` update required us to update `tensorflow`, or roll back. I decided to take a chance.

I therefore updated the source build, and it builds just fine on x86_64-linux (with and without cuda).
The binary builds otoh fail, because upstream does not provide wheels for 2.8.0 on MacOS. The links do exist, but they 404.

Obviously this is yet another release with an absurd number of CVEs :facepalm: 

https://github.com/tensorflow/tensorflow/releases/tag/v2.8.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux (with and without cuda)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/tensorflow/default.nix [hunk 1]
```diff
@@ -361,12 +356,12 @@ let
     fetchAttrs = {
       # cudaSupport causes fetch of ncclArchive, resulting in different hashes
       sha256 = if cudaSupport then
-        "sha256-+szc2mRoImwijzbj3nw6HmZp3DeRjjPRU5yC+5AEbkg="
+        "sha256-dQEyfueuQPcGvbhuh8Al45np3nRLDw2PCfC2lEqAH50="
       else
         if stdenv.isDarwin then
-          "sha256-+bwIzp6t7gRJPcI8B5oyuf9z0AjCAyggUR7x+vv5kFs="
+          "0000000000000000000000000000000000000000000000000000000000000000"
```
Comment (mweinelt) [inline pkgs/development/python-modules/tensorflow/default.nix:362]: I hope ofBorg can help me out here!

**Comment by mweinelt** [comment]
@ofBorg build python3Packages.tensorflow

File: pkgs/development/python-modules/tensorflow/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/tensorflow/default.nix:362]: If that doesn't work I have to fall back to use the bundled version.

**Comment by mweinelt** [comment]
Actually tensorflowWithCuda requires nsync, so excluding it only on darwin.

**Comment by jonringer** [comment]
> Obviously this is yet another release with an absurd number of CVEs facepalm

They didn't adhere to C++'s `#1` rule, you just need to write perfect code every time.

**Pull request merged**